src/generators/typescript/presets/utils/ExampleFunction.ts

```diff
@@ -0,0 +1,78 @@
+import { TypeScriptRenderer } from '../../TypeScriptRenderer';
+import { CommonModel } from '../../../../models';
+
+/**
+ * LAZY implementation, find first acceptable value
+ * 
+ * @param model 
+ */
+export function renderValueFromModel(model: CommonModel, renderer: TypeScriptRenderer): string | undefined {
+  if (Array.isArray(model.enum) && model.enum.length > 0) {
+    return JSON.stringify(model.enum[0]);
+  }
+  if (model.$ref !== undefined) {
+    return `${renderer.nameType(model.$ref)}.example()`;
+  }
+  if (Array.isArray(model.type)) {
+    if (model.type.length > 0) {
+      return renderValueFromType(model.type[0], model, renderer);
+    } 
+    return undefined;
+  }
+  return renderValueFromType(model.type, model, renderer);
+}
+
+export function renderValueFromType(modelType: string | undefined, model: CommonModel, renderer: TypeScriptRenderer): string | undefined {
+  if (modelType === undefined) {
+    return undefined;
+  }
+  switch (modelType) {
+  case 'string':
+    return '"string"';
+  case 'integer':
+  case 'number':
+    return '0';
+  case 'boolean':
+    return 'true';
+  case 'array': {
+    if (model.items === undefined) {
+      return '[]';
+    }
+    //Check and see if it should be rendered as tuples 
+    if (Array.isArray(model.items)) {
+      const arrayValues = model.items.map((item) => {
+        return renderValueFromModel(item, renderer);
+      });
+      return `[${arrayValues.join(', ')}]`;
+    }
+    const arrayType = renderValueFromModel(model.items, renderer);
+    return `[${arrayType}]`;
+  }
+  }
+  return undefined;
+}
+/**
+ * Render `example` function based on model properties
+ */
+export default function renderExampleFunction({ renderer, model }: {
+  renderer: TypeScriptRenderer,
+  model: CommonModel,
+}): string {
+  const properties = model.properties || {};
+  const setProperties = [];
+  for (const [propertyName, property] of Object.entries(properties)) {
```

Comment:
What about `patternProperties` and `additionalProperties`? Will be added in the followup?

Reply:
I have no use-case for this, which is also why it is a lazy implementation, find the first example representation you can and use that 🙂 That being said, if others find the need they can of course adapt it 😄

Reply:
Yeah, we can add support for this (as option) if someone will raise that issue.

Reply:
Yea exactly 👍